src/orange/orangeinp/CsgTreeIO.json.cc

```diff
@@ -129,11 +129,12 @@ void to_json(nlohmann::json& j, CsgUnit const& unit)
     }();
 
     j["volumes"] = [&unit] {
-        CELER_ASSERT(unit.volumes.size() == unit.fills.size());
+        CELER_ASSERT(unit.tree.volumes().size() == unit.fills.size());
```

Comment:
Do you mean that you'd have 
```json
{
"tree:" { "nodes": [], "volumes": []}
}
```
instead of the "tree" key being a list?

Reply:
mmh, an entry for a volume needs material data from the CsgUnit which we can't pass to the `to_json` serialization function

Reply:
With the following changes, many tests fail because of the new structure. This would require many changes. It may be better to do that in a separate PR.
```diff
diff --git a/src/orange/orangeinp/CsgTreeIO.json.cc b/src/orange/orangeinp/CsgTreeIO.json.cc
index f355328c1..f3ec75bf1 100644
--- a/src/orange/orangeinp/CsgTreeIO.json.cc
+++ b/src/orange/orangeinp/CsgTreeIO.json.cc
@@ -65,11 +65,21 @@ void to_json(json& j, CsgTree const& tree)
 {
     ContainerVisitor<CsgTree const&, CsgTree::NodeId> visit_tree{tree};
 
-    j = json::array();
+    j = json::object();
+
+    json nodes = json::array();
     for (auto n : range(CsgTree::NodeId{tree.size()}))
     {
-        j.push_back(visit_tree(JsonConverter{}, n));
+        nodes.push_back(visit_tree(JsonConverter{}, n));
+    }
+    j["nodes"] = nodes;
+
+    json volumes = json::array();
+    for (auto id : tree.volumes())
+    {
+        volumes.push_back(id.get());
     }
+    j["volumes"] = volumes;
 }
 
 namespace detail
@@ -128,18 +138,19 @@ void to_json(nlohmann::json& j, CsgUnit const& unit)
         return result;
     }();
 
-    j["volumes"] = [&unit] {
-        CELER_ASSERT(unit.tree.volumes().size() == unit.fills.size());
-        json result = json::array();
-        for (auto i : range(unit.tree.volumes().size()))
+    j["fills"] = [&volumes = unit.tree.volumes(), &fills = unit.fills] {
+
+        CELER_ASSERT(volumes.size() == fills.size());
+        json result = json::object();
+        for (auto const& i : range(fills.size()))
         {
-            auto entry
-                = nlohmann::json{{"csg_node", unit.tree.volumes()[i].get()}};
-            if (auto* m = std::get_if<GeoMaterialId>(&unit.fills[i]))
+
+            auto entry = json::object();
+            if (auto* m = std::get_if<GeoMaterialId>(&fills[i]))
             {
                 entry["material"] = m->unchecked_get();
             }
-            else if (auto* d = std::get_if<Daughter>(&unit.fills[i]))
+            else if (auto* d = std::get_if<Daughter>(&fills[i]))
             {
                 entry["universe"] = d->universe_id.unchecked_get();
                 entry["transform"] = d->transform_id.unchecked_get();
@@ -149,15 +160,6 @@ void to_json(nlohmann::json& j, CsgUnit const& unit)
         return result;
     }();
 
-    j["fills"] = [&volumes = unit.tree.volumes()] {
-        json result = json::array();
-        for (auto const& v : volumes)
-        {
-            result.push_back(v.unchecked_get());
-        }
-        return result;
-    }();
-
     if (unit.background)
     {
         j["background"] = unit.background.get();
```